backend/substrapp/views/utils.py

```diff
@@ -143,43 +143,11 @@ def _download_remote_file(self, channel_name, ledger_field, asset):
         return response
 
 
-def find_primary_key_error(validation_error, key_name='pkhash'):
-    detail = validation_error.detail
-
-    def find_unique_error(detail_dict):
-        for key, errors in detail_dict.items():
-            if key != key_name:
-                continue
-            for error in errors:
-                if error.code == 'unique':
-                    return error
-
-        return None
-
-    # according to the rest_framework documentation,
-    # validation_error.detail could be either a dict, a list or a nested
-    # data structure
-
-    if isinstance(detail, dict):
-        return find_unique_error(detail)
-    elif isinstance(detail, list):
-        for sub_detail in detail:
-            if isinstance(sub_detail, dict):
-                unique_error = find_unique_error(sub_detail)
-                if unique_error is not None:
-                    return unique_error
-
-    return None
-
-
 def validate_pk(pk):
-    if len(pk) != 64:
-        raise exceptions.BadRequestError(f'Wrong pk {pk}')
-
     try:
-        int(pk, 16)  # test if pk is correct (hexadecimal)
+        uuid.UUID(pk)
```

Comment:
do we still need to validate_pk?

Reply:
This function helps in the sense that it provides a consistent HTTP response for invalid input keys. 

Without this function, the key would be propagated to a ledger invoke, a ledger query, a DB call etc, resulting in different error types. I tried running the unit tests without this function and they fail

```
AttributeError: 'Settings' object has no attribute 'LEDGER_CALL_RETRY'
```

That's because we try to do an invoke using an invalid key.

Note: I renamed the function to `validate_key` because it makes more sense than `validate_pk` IMO https://github.com/SubstraFoundation/substra-backend/pull/342/commits/41738f71c316fe6866798bada0d3bbefbe6c9628